src/processes.c

```diff
@@ -2319,23 +2194,23 @@ static int ps_read (void)
 			 */
 			if (procs[i].ki_flag & P_INMEM)
 			{
-				pse.cpu_user_counter = procs[i].ki_rusage.ru_utime.tv_usec
+				pse.counters.cpu_user = procs[i].ki_rusage.ru_utime.tv_usec
 				       	+ (1000000lu * procs[i].ki_rusage.ru_utime.tv_sec);
-				pse.cpu_system_counter = procs[i].ki_rusage.ru_stime.tv_usec
+				pse.counters.cpu_system = procs[i].ki_rusage.ru_stime.tv_usec
 					+ (1000000lu * procs[i].ki_rusage.ru_stime.tv_sec);
 			}
 
 			/* no I/O data */
-			pse.io_rchar = -1;
-			pse.io_wchar = -1;
-			pse.io_syscr = -1;
-			pse.io_syscw = -1;
-			pse.io_diskr = -1;
-			pse.io_diskw = -1;
+			pse.gauges.io_rchar = -1;
```

Comment:
I wonder if this can be accomplished by simply assigning `procstat_gauges_init` to `pse.gauges` before reading the values... In fact, can we do this right after we declare `pse`?

Reply:
Same concerns as above, I'm afraid to make a sweeping set of assumptions when "defaults" vary.